numba/core/interpreter.py

```diff
@@ -892,6 +897,114 @@ def op_BUILD_TUPLE_UNPACK_WITH_CALL(self, inst, tuples, temps):
+        # the rest of the bytecode refers to the build_list var so the result
+        # of actually building this list and making it into a tuple needs to go
+        # into that var, which means the original one needs to be something
+        # else!
+        new_build_list = self.store(init.target, res)
+        acc = new_build_list
+        for x in peep_hole:
+            if isinstance(x, ir.Assign):
+                if isinstance(x.value, ir.Expr):
+                    expr = x.value
+                    if expr.op == 'getattr':
+                        if expr.attr in ('extend', 'append'):
+                            # don't want these
+                            continue
+                    elif expr.op == 'call':
+                        fname = expr.func.name
+                        if fname in extends or fname in appends:
+                            arg = expr.args[0]
+                            if isinstance(arg, ir.Var):
+                                tmp_name = "%s_var_%s" % (fname, arg.name)
+                                if fname in appends:
+                                    bt = ir.Expr.build_tuple(items=[arg,], loc=expr.loc)
+                                else:
+                                    bt = arg
+                                var = ir.Var(arg.scope, tmp_name, expr.loc)
+                                asgn = ir.Assign(bt, var, expr.loc)
+                                local_store(asgn)
+                                arg = var
+
+                            # this needs to be a binary add
+                            new = ir.Expr.binop(fn=operator.add,
+                                                lhs=acc,
+                                                rhs=arg,
+                                                loc=x.loc)
+                            asgn = ir.Assign(new, x.target, expr.loc)
+                            local_store(asgn)
+                            acc = asgn.target
+                    elif expr.op == 'build_list':
+                        new = ir.Expr.build_tuple(expr.items, expr.loc)
+                        asgn = ir.Assign(new, new_build_list, expr.loc)
+                        # Not a temporary any more
+                        self.assigner.unused_dests.remove(new_build_list.name)
+                        local_store(asgn)
+                        del self.definitions[x.target.name]
+                    else:
+                        local_store(x)
+                else:
+                    local_store(x)
+
+            else:
+                # stick everything else in as-is
+                local_store(x)
+        # write the result back into the original build list as everything
+        # refers to it.
+        local_store(ir.Assign(acc, the_build_list, the_build_list.loc))
+
+        cpy = self.current_block.body[:]
+        tmp = cpy[:len(cpy) -count -2]
+        tmp.extend(new_hole)
+        self.current_block.body = tmp
```

Comment:
I wanted to move the peephole into a post-processing immediately after the bytecode->IR. Doing it here makes it hard to debug and the IR mutation here is not ideal.

Reply:
Agree, now we know this works, will delay peephole rewrites to the point where there's actual tools available to help with doing it!